core/peer/peer.go

```diff
@@ -808,3 +794,109 @@ func (p *PeerImpl) GetTransactionResultByUUID(txUuid string) (*pb.TransactionRes
 	defer p.ledgerWrapper.RUnlock()
 	return p.ledgerWrapper.ledger.GetTransactionResultByUUID(txUuid)
 }
+
+// initDiscovery load the addresses from the discovery list previously saved to disk and adds them to the current discovery list
+func (p *PeerImpl) initDiscovery() []string {
+	p.discHelper = discovery.NewDiscoveryImpl()
+	p.discPersist = viper.GetBool("peer.discovery.persist")
+	if !p.discPersist {
+		peerLogger.Warning("Discovery list will not be persisted to disk")
```

Comment:
I claim that it should not because you may run the peer with persistence and then decide for whatever reason to switch it off at a later point. Let me know if you disagree. Also note that at some point we'll need to develop a workflow that removes nodes from the discovery list. (The method is already there, but the process that will result in this invocation has not been defined yet.)